robusta_krr/core/integrations/prometheus/metrics_service/prometheus_metrics_service.py

```
@@ -104,31 +106,43 @@ def __init__(
-        self.prom_config = generate_prometheus_config(url=self.url, headers=headers, metrics_service=self)
-        self.prometheus = get_custom_prometheus_connect(self.prom_config)
+        self.headers = headers
+        self.prom_config = None
+        self.prometheus = None
+        self.get_prometheus()
+
+    def get_prometheus(self):
+        now = datetime.utcnow()
+        if (not self.prometheus
+            or not self._last_init_at
+            or now - self._last_init_at >= timedelta(seconds=PROM_REFRESH_CREDS_SEC)):
```

Flake8 (7.3.0)

[error] visually indented line with same indent as next logical line

(E129)